Library/Homebrew/test/livecheck/strategy/sparkle_spec.rb

```diff
@@ -7,92 +7,178 @@
+        <enclosure os="macos" url="#{item_hashes[:version_120][:url]}" sparkle:shortVersionString="#{item_hashes[:version_120][:short_version]}" sparkle:version="#{item_hashes[:version_120][:version]}" length="12345678" type="application/octet-stream" sparkle:dsaSignature="ABCDEF+GHIJKLMNOPQRSTUVWXYZab/cdefghijklmnopqrst/uvwxyz1234567==" />
+      </item>
+    EOS
+
+    # This is the full 'normal' XML Feed
+    appcast = appcast_xml_feed_string <<~EOS
+      #{version_123_item}
+      #{version_122_item}
+      #{version_121_item_with_osx_os}
+      #{version_120_item_with_macos_os}
+    EOS
+
+    # This is the same 'normal' appcast feed with an additional set of
+    # items to be filtered out. It's:
+    #
+    # 1. The first item with a bad `os`
+    # 1. The first item with one bad minimum system version
+    # 1. The first item with another bad minimum system version
+    # 1. An empty item
+    #
+    # The rest is identical to the original appcast feed.
+    omitted_items = appcast_xml_feed_string <<~EOS
+      #{version_123_item}
+      #{version_123_item.sub(%r{<(enclosure[^>]+?)\s*?/>}, '<\1 os="not-osx-or-macos" />')}
+      #{version_123_item.sub(/(<sparkle:minimumSystemVersion>)[^<]+?</m, '\1100<')}
+      #{version_123_item.sub(/(<sparkle:minimumSystemVersion>)[^<]+?</m, '\19000<')}
+      <item>
       </item>
+      #{version_122_item}
+      #{version_121_item_with_osx_os}
+      #{version_120_item_with_macos_os}
     EOS
 
-    second_item = <<~EOS
+    beta_channel_item_string = <<~EOS
       <item>
-        <title>#{item_hash[1][:title]}</title>
+        <title>#{item_hashes[:version_123][:title]}</title>
+        <sparkle:channel>beta</sparkle:channel>
         <sparkle:minimumSystemVersion>10.10</sparkle:minimumSystemVersion>
-        <sparkle:releaseNotesLink>https://www.example.com/example/#{item_hash[1][:short_version]}.html</sparkle:releaseNotesLink>
-        <pubDate>#{item_hash[1][:pub_date]}</pubDate>
-        <sparkle:version>#{item_hash[1][:version]}</sparkle:version>
-        <sparkle:shortVersionString>#{item_hash[1][:short_version]}</sparkle:shortVersionString>
-        <link>#{item_hash[1][:url]}</link>
+        <sparkle:releaseNotesLink>https://www.example.com/example/#{item_hashes[:version_123][:short_version]}.html</sparkle:releaseNotesLink>
+        <pubDate>#{item_hashes[:version_123][:pub_date]}</pubDate>
+        <enclosure url="#{item_hashes[:version_123][:url]}" sparkle:shortVersionString="#{item_hashes[:version_123][:short_version]}" sparkle:version="#{item_hashes[:version_123][:version]}" length="12345678" type="application/octet-stream" sparkle:dsaSignature="ABCDEF+GHIJKLMNOPQRSTUVWXYZab/cdefghijklmnopqrst/uvwxyz1234567==" />
       </item>
     EOS
+    beta_channel_item = appcast_xml_feed_string <<~EOS
+      #{beta_channel_item_string}
+      #{version_122_item}
+      #{version_121_item_with_osx_os}
+      #{version_120_item_with_macos_os}
+    EOS
```

Comment:
I feel a little more strongly about this than I do about the other changes that I'm accepting.

I find the approach of just explicitly building up the string you expect rather than starting with the existing `appcast` string and substituting a substitution of `version_123_item` into it requires a lot of contextual awareness and introduces coupling between `appcast` and `beta_channel_item`.

I think it's significantly cleaner (but also slightly WETter) to just build up the `beta_channel_item` directly from the components. This decouples `appcast` and `beta_channel_item` and, to my eyes at least, reads much more clearly.

For now I'd like to keep this but please let me know if you still feel strongly that it should be changed.